tests/UuidBinaryOrderedTimeTypeTest.php

```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    protected function setUp()
+    {
+        $this->platform = $this->getPlatformMock();
+        $this->platform->expects($this->any())
+            ->method('getBinaryTypeDeclarationSQLSnippet')
+            ->will($this->returnValue('DUMMYBINARY(16)'));
+
+        $this->type = Type::getType('uuid_binary_ordered_time');
+    }
+
+    public function testGetName()
+    {
+        $this->assertEquals('uuid_binary_ordered_time', $this->type->getName());
+    }
+
+    public function testUuidConvertsToDatabaseValue()
+    {
+        $uuid = Uuid::fromString('ff6f8cb0-c57d-11e1-9b21-0800200c9a66');
+
+        $expected = hex2bin('11e1c57dff6f8cb09b210800200c9a66');
+        $actual = $this->type->convertToDatabaseValue($uuid, $this->platform);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testStringUuidConvertsToDatabaseValue()
+    {
+        $uuid = 'ff6f8cb0-c57d-11e1-9b21-0800200c9a66';
+
+        $expected = hex2bin('11e1c57dff6f8cb09b210800200c9a66');
+        $actual = $this->type->convertToDatabaseValue($uuid, $this->platform);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testInvalidUuidConversionForDatabaseValue()
+    {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
+        $this->type->convertToDatabaseValue('abcdefg', $this->platform);
+    }
+
+    public function testNullConversionForDatabaseValue()
+    {
+        $this->assertNull($this->type->convertToDatabaseValue(null, $this->platform));
+    }
+
+    public function testUuidConvertsToPHPValue()
+    {
+        $uuid = $this->type->convertToPHPValue(hex2bin('11e1c57dff6f8cb09b210800200c9a66'), $this->platform);
+        $this->assertInstanceOf('Ramsey\Uuid\Uuid', $uuid);
+        $this->assertEquals('ff6f8cb0-c57d-11e1-9b21-0800200c9a66', $uuid->toString());
+    }
+
+    public function testInvalidUuidConversionForPHPValue()
+    {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
+        $this->type->convertToPHPValue('abcdefg', $this->platform);
+    }
+
+    public function testUnsupportedUuidConversionToDatabaseValue()
+    {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
+        $this->type->convertToDatabaseValue(Uuid::uuid4(), $this->platform);
+    }
+
+    /**
+     * @dataProvider provideUnsupportedDatabaseValues
+     * @param string $databaseValue
+     */
+    public function testUnsupportedUuidConversionToPHPValue($databaseValue)
+    {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
+        $this->type->convertToPHPValue(hex2bin($databaseValue), $this->platform);
+    }
+
+    public function testNullConversionForPHPValue()
+    {
+        $this->assertNull($this->type->convertToPHPValue(null, $this->platform));
+    }
+
+    public function testReturnValueIfUuidForPHPValue()
+    {
+        $uuid = Uuid::uuid4();
```

Comment:
Should not this be ``uuid1()``? The result would not be different, since you are just returning it, but it would keep consistency with your check on ``assertUuidV1()`` (even if you are not checking if an Uuid instance is passed)